src/circularbarn.cpp

```diff
@@ -0,0 +1,38 @@
+#include <bits/stdc++.h>
+
+using namespace std;
+
+int main() {
+    ios_base::sync_with_stdio(false);
+    cin.tie(nullptr);
+
+    freopen("cbarn.in", "r", stdin);
+    freopen("cbarn.out", "w", stdout);
+
+    int numRooms;
+    cin >> numRooms;
+    vector<int> numCowInRoom;
+    for (int i = 0; i < numRooms; ++i) {
+        int numCows;
+        cin >> numCows;
+        numCowInRoom.emplace_back(numCows);
+    }
+
+    int maxNumMoves = INT_MAX;
+    for (int i = 0; i < numRooms; ++i) {
+        int numMoves = 0;
+        for (int j = 0; j < numRooms - i; ++j) {
```

Comment:
Logic of Line24~Line30 can be combined using the idea of modulo (mod by `numRooms`)